engine/executor/server.go

```diff
@@ -640,31 +641,20 @@ func (s *Server) keepHeartbeat(ctx context.Context) error {
 				// executor actually wait for a timeout when ttl is nearly up.
 				Ttl: uint64(s.cfg.KeepAliveTTL.Milliseconds() + s.cfg.RPCTimeout.Milliseconds()),
 			}
-			resp, err := s.masterClient.Heartbeat(ctx, req)
+			_, err := s.masterClient.Heartbeat(ctx, req)
 			if err != nil {
 				log.L().Error("heartbeat rpc meet error", zap.Error(err))
+				// TODO: move all errors into one package, so that executor
+				// server no longer needs to depend on servermaster package.
+				if servermaster.ErrTombstoneExecutorError.Is(err) {
+					return errors.ErrHeartbeat.GenWithStack("logic error: %v", err)
+				}
 				if s.lastHearbeatTime.Add(s.cfg.KeepAliveTTL).Before(time.Now()) {
-					return errors.WrapError(errors.ErrHeartbeat, err, "rpc")
+					return errors.WrapError(errors.ErrHeartbeat, err, "timeout")
 				}
 				continue
 			}
-			if resp.Err != nil {
-				log.L().Warn("heartbeat response meet error", zap.Stringer("code", resp.Err.GetCode()))
-				switch resp.Err.Code {
-				case pb.ErrorCode_UnknownExecutor:
-					log.L().Info("heartbeat failed, will retry", zap.Error(err))
-					continue
-				case pb.ErrorCode_TombstoneExecutor:
-					return errors.ErrHeartbeat.GenWithStack("logic error: %s", resp.Err.GetMessage())
-				case pb.ErrorCode_MasterNotReady:
-					s.lastHearbeatTime = t
-					if rl.Allow() {
-						log.L().Info("heartbeat success with MasterNotReady")
-					}
```

Comment:
This case is not checked after this PR.

Reply:
Addressed in https://github.com/pingcap/tiflow/pull/7362/commits/7dc5da1bf53b9fd03abf0373f8b120380f6b7d53.